### ISSUE TITLE
Database of monsters

### ISSUE THREAD

**Issue by chefranov** [fonsleenaars/tqdb#22]
Is it possible to create DB of monsters? It would be cool.

**Comment by fonsleenaars** [comment]
Definitely possible, what kind of setup are you looking for? A list of monsters with their skills, statistics, etc?

More details = better!

To be fair though, this won't have priority in the immediate future, but after all the bugs have been fixed and item database things are implemented, we can begin on the monster list.

**Comment by chefranov** [comment]
Thank you for your reply. 
A list with monsters shoud include:
- image of monster
- name
- skills
- statistics depend on difficulty of game
- where we can find this monster
- somthing else

**Comment by LordVampir1983** [comment]
**website**
It would also be useful to have references to charms, which they can drop off. And links back, for easy navigation

**Comment by fonsleenaars** [comment]
Sure, charms & relics are already indexed so the reverse index will be simple to add.

I think the Monster Database will just be a new menu item (I think I'll reorganize the menu once we're there, looks like we're expanding quite a bit) and it'll have a similar look and feel to the current database.

That way navigating back and forth will be simple & smooth.

**Comment by TheImmorthal** [comment]
Hei. Any news about this database of monsters? I would be really glad if we would have one.

**Comment by fonsleenaars** [comment]
Some news, after 1.2.0 is released I'm adding the monster database. It's a little bit of a mess because a lot of monsters have properties that are messing up the parser a little bit (weird things like 1% Lightning Resistance, who cares about that!)

It shouldn't be TOO much longer, but I'll be able to give you a better estimate in a week or so

**Comment by TheImmorthal** [comment]
Hello. I'm glad work is in progress. Well about those 1% resistances or so they can be forgotten and focus on the important things

    On Friday, July 6, 2018, 10:11:36 AM GMT+2, Fons Leenaars <notifications@github.com> wrote:  
 
 
Some news, after 1.2.0 is released I'm adding the monster database. It's a little bit of a mess because a lot of monsters have properties that are messing up the parser a little bit (weird things like 1% Lightning Resistance, who cares about that!)

It shouldn't be TOO much longer, but I'll be able to give you a better estimate in a week or so

**Comment by fonsleenaars** [comment]
Agreed, the tricky part is that I have to know when to skip those properties and when they can be important, unfortunately I can't just tell the parser "if it's a silly property, skip it".

But I've made some progress. I'm not sure yet how I'll handle the images for monsters but we'll get to that later, indexing them by name and properties comes first.

**Comment by TheImmorthal** [comment]
Of course you can also forget about the images of monsters at beginning and focusing on that later bro. 
    On Friday, July 6, 2018, 1:15:39 PM GMT+2, Fons Leenaars <notifications@github.com> wrote:  
 
 
Agreed, the tricky part is that I have to know when to skip those properties and when they can be important, unfortunately I can't just tell the parser "if it's a silly property, skip it".

But I've made some progress. I'm not sure yet how I'll handle the images for monsters but we'll get to that later, indexing them by name and properties comes first.

**Comment by fonsleenaars** [comment]
I've finished the first preview: https://staging.tq-db.net/monsters

The first setup has the following details per monster:

- Properties
  - HP/MP
  - Damage (right click attack)
  - Defensive resistances
- Abilities
  - Regular skills
  - Summons
  - Some misc. skills that need to be fixed (defensive res. skills that should just be added to primary stats)
- Loot table

Then these monsters are indexed by difficulty as well (at the bottom it has Normal/Epic/Legendary).

There's still quite a bit of work to do here, but this first setup will be deployed pretty soon to the live website. Just wanted to keep you guys updated!

**Comment by chefranov** [comment]
Awesome! How about images of monsters? Is it possible?

**Comment by fonsleenaars** [comment]
I'm not sure, I have absolutely no experience with 3D modeling and after some quick research I see that creatures only have the mesh and texture files available. If I'm not mistaken this is because the mesh can be viewed in different angles/positions (which makes sense for in-game) but I'm not sure how I can go about converting a mesh with a bunch of textures into something static like a JPG of PNG.

I'll look around online and see if I can find some tools that can do so. If it ends up being possible, I'll add them for sure.

**Comment by chefranov** [comment]
@fonsleenaars Titan Quest has tool Viewer.exe where we can see 3D models. Maybe it can help you.

**Comment by fonsleenaars** [comment]
Thanks, I've been looking into that tool but I can't seem to get the list of available models loaded, once I have it working I'll see if I can somehow export a camera-locked image from there.

**Comment by chefranov** [comment]
@fonsleenaars thanks for you work! 
Can you add information about location of monsters? TQ DB contains this information?
And I have some ideas about new features for your site/project. I can write about it in other topic.

**Comment by fonsleenaars** [comment]
I'll see what I can do about locations, I don't believe the database record files contain any information about that, but perhaps some other resources will.

**Comment by fonsleenaars** [comment]
@chefranov the monster database has been live for a while now, closing this one. Locations are still the tricky part, the game files don't really say where monsters spawn, so no promises on that 😅

**Comment by chefranov** [comment]
I already done this https://titanquest.org.ua/tqdb#/creatures

**Comment by chefranov** [comment]
A little bit screens of a new desgin. 

<img width="1181" height="638" alt="Image" src="https://github.com/user-attachments/assets/676d96c7-2379-42cf-97dd-5b3fb189f49f" />

<img width="1170" height="1321" alt="Image" src="https://github.com/user-attachments/assets/d28c9d80-0cd4-4ad4-b7cb-5ecd25c4adcf" />

Also I can render 3d in browsers

https://github.com/user-attachments/assets/1c3dea69-3c57-4b35-bc90-601310a531f7
